src/devices/sound/hc55516.cpp

```diff
@@ -293,16 +566,30 @@ void hc55516_device::sound_stream_update(sound_stream &stream, std::vector<read_
 
 	/* software driven clock */
 	else
-		for (i = 0; i < buffer.samples(); i++, sample += slope)
+		for (int i = 0; i < buffer.samples(); i++, sample += slope)
 			buffer.put(i, stream_buffer::sample_t(sample));
 }
 
-void mc3417_device::sound_stream_update(sound_stream &stream, std::vector<read_stream_view> const &inputs, std::vector<write_stream_view> &outputs)
+
+
+//##########################################
+//                 MC3418
+//##########################################
+DEFINE_DEVICE_TYPE(MC3418, mc3418_device, "mc3418", "MC3418")
+
+mc3418_device::mc3418_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: mc3417_device(mconfig, MC3418, tag, owner, clock)
+{
+}
+
+/*
+void mc3417_device::sound_stream_update_legacy(sound_stream &stream, stream_sample_t const * const *inputs, stream_sample_t * const *outputs, int samples)
 {
-	hc55516_device::sound_stream_update(stream, inputs, outputs);
+	cvsd_device_base::sound_stream_update_legacy(stream, inputs, outputs, samples);
 }
 
-void mc3418_device::sound_stream_update(sound_stream &stream, std::vector<read_stream_view> const &inputs, std::vector<write_stream_view> &outputs)
+void mc3418_device::sound_stream_update_legacy(sound_stream &stream, stream_sample_t const * const *inputs, stream_sample_t * const *outputs, int samples)
 {
-	hc55516_device::sound_stream_update(stream, inputs, outputs);
+	cvsd_device_base::sound_stream_update_legacy(stream, inputs, outputs, samples);
 }
+*/
```

Comment:
I don’t really see the point of keeping these commented base class calls here.

Reply:
Removed, they were old leftovers from before squashing the audio implementations for the harris vs motorola chips together into groups of subclasses.